modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -37,54 +40,134 @@ cv::gimpl::GExecutor::GExecutor(std::unique_ptr<ade::Graph> &&g_model)
+    for (auto&& nh : ade::util::filter(sorted.nodes(), is_slot)){
+        const auto orig_data_nh
+            = m_gim.metadata(nh).get<DataSlot>().original_data_node;
+        // (1)
+        initResource(nh, orig_data_nh);
+        m_slots.emplace_back(DataDesc{nh, orig_data_nh});
+    }
 
-        case NodeKind::SLOT:
-            {
-                const auto orig_data_nh
-                    = m_gim.metadata(nh).get<DataSlot>().original_data_node;
-                // (1)
-                initResource(nh, orig_data_nh);
-                m_slots.emplace_back(DataDesc{nh, orig_data_nh});
-            }
-            break;
+    namespace r = ade::util::Range;
+    for (auto&& nh : r::filter(sorted.nodes(), is_island)){
+        std::vector<RcDesc> input_rcs;
+        std::vector<RcDesc> output_rcs;
+        input_rcs.reserve(nh->inNodes().size());
+        output_rcs.reserve(nh->outNodes().size());
+
+        auto xtract = [&](ade::NodeHandle slot_nh, std::vector<RcDesc> &vec) {
+            const auto orig_data_nh
+                = m_gim.metadata(slot_nh).get<DataSlot>().original_data_node;
+            const auto &orig_data_info
+                = m_gm.metadata(orig_data_nh).get<Data>();
+            vec.emplace_back(RcDesc{ orig_data_info.rc
+                                   , orig_data_info.shape
+                                   , orig_data_info.ctor});
+        };
+        // (3)
+        for (auto in_slot_nh  : nh->inNodes())  xtract(in_slot_nh,  input_rcs);
+        for (auto out_slot_nh : nh->outNodes()) xtract(out_slot_nh, output_rcs);
+
+        m_ops.emplace_back(OpDesc{ std::move(input_rcs)
+                                 , std::move(output_rcs)
+                                 , m_gim.metadata(nh).get<IslandExec>().object
+                                 });
+    }
+
+}
+
+#if defined(USE_GAPI_TBB_EXECUTOR)
+cv::gimpl::GTBBExecutor::GTBBExecutor(std::unique_ptr<ade::Graph> &&g_model)
+    : GExecutor(std::move(g_model))
+{
+    using op_index_t = size_t;
+    std::unordered_map<ade::NodeHandle, op_index_t, ade::HandleHasher<ade::Node>> op_indexes;
+    std::unordered_map<op_index_t, std::unordered_set<ade::NodeHandle, ade::HandleHasher<ade::Node>>> op_dependees_node_handles;
+
+    namespace r = ade::util::Range;
+    auto is_island = [&](ade::NodeHandle const& nh){
+        return m_gim.metadata(nh).get<NodeKind>().k == NodeKind::ISLAND;
+    };
+
+    auto sorted = m_gim.metadata().get<ade::passes::TopologicalSortData>();
+
+    for (auto&& indexed_nh : r::indexed(r::filter(sorted.nodes(), is_island))){
+        auto&& nh           = r::value(indexed_nh);
+        const auto op_index = r::index(indexed_nh);
+
+        op_indexes[nh] = op_index;
+
+        //TODO: use async tasks for non CPU islands
+        std::function<void()> body = [this, op_index](){
+            GExecutor::run_op(m_ops[op_index], m_res);
+        };
+        tasks.emplace_back(std::move(body));
+
+        auto dependents_range = r::map(
```

Comment:
This should be named vice versa, it's the map of the current operation's dependencies, not dependents, right?

Reply:
thanks - fixed